manifests/init.pp

```diff
@@ -1,24 +1,19 @@
-# -- Class systemd  
-# This module allows triggering systemd commands once for all modules 
+# This module allows triggering systemd commands once for all modules
+#
+# @api public
+#
+# @param service_limits
+#   May be passed a resource hash suitable for passing directly into the
+#   ``create_resources()`` function as called on ``systemd::service_limits``
+#
 class systemd (
-  $service_limits = {}
+  Optional[Hash] $service_limits = undef
```

Comment:
I'd like to leave this one if possible. The compiler is so bloated already that everything we can bind to a single symbol helps.

Also, the code is easier to read and not confusing if, for some reason, we decide to include data in modules later.